ledger/streaming/api/src/convert/components.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Convert between Rust and proto representations of BlockStreamComponents.
+
+use crate::{streaming_blocks::BlockWithQuorumSet, BlockStreamComponents};
+use mc_api::ConversionError;
+use mc_attest_core::VerificationReport;
+use mc_consensus_scp::QuorumSet;
+use std::convert::{TryFrom, TryInto};
+
+impl From<&BlockStreamComponents> for BlockWithQuorumSet {
+    fn from(data: &BlockStreamComponents) -> Self {
+        let mut proto = BlockWithQuorumSet::new();
+        proto.set_block((&data.block_data).into());
+        if let Some(ref quorum_set) = data.quorum_set {
+            proto.set_quorum_set(quorum_set.into());
+        }
+        if let Some(ref report) = data.verification_report {
+            proto.set_report(report.into());
+        }
+        proto
+    }
+}
+
+impl TryFrom<&BlockWithQuorumSet> for BlockStreamComponents {
+    type Error = ConversionError;
```

Comment:
Should we merge our error set? Or does an error set for each crate make sense? I think putting some thought into separating by ontology of errors + their handling is a worthy discussion.

Reply:
We do have an `mc_ledger_streaming_api::Error`, but `mc_api::ConversionError` seems sufficient here, and matches the other conversion code in `mc_api::convert`. `ConversionError` is already an enum, so I expect Chris' work won't affect this.

Reply:
Also note that `mc_ledger_streaming_api::Error` has an enum variant wrapping `ConversionError`

Reply:
It sounds like there's no reason *not* to use `mc_ledger_streaming_api::Error`, then.

I'm fine to leave this here now, but only because we need to refactor *all* our proto stuff to make it easier for non-rust clients to use, and that's a good time to split up our monolithic API crates into smaller/more manageable.